extensions/positron-catalog-explorer/src/catalogs/snowflake.ts

```diff
@@ -73,195 +66,223 @@ export function registerSnowflakeProvider(
-
-		// User is not re-authenticating an existing account, prompt for account name
-		if (!account) {
-			// If we have recent accounts, show a quick pick first
-			if (recentAccounts.length > 0) {
-				// Add a "New Account" option at the end
-				const items = [
-					...recentAccounts.map(name => ({
-						label: name,
-						description: 'Recent account'
-					})),
-					{
-						label: 'Enter New Account...',
-						description: 'Provide a different account identifier'
-					}
-				];
-
-				const selection = await vscode.window.showQuickPick(items, {
-					placeHolder: 'Select a recent account or enter a new one',
-					ignoreFocusOut: true
-				});
-
-				if (!selection) {
-					return undefined; // User canceled
+	// Load all connections from connections.toml
+	const connections = await getSnowflakeConnectionOptions();
+
+	// User is not re-authenticating an existing connection, prompt for one
+	if (!connName) {
+		let items: vscode.QuickPickItem[] = [];
+
+		if (connections && Object.keys(connections).length > 0) {
+			// If we found connections, add them to the quick pick
+			const connectionNames = Object.keys(connections);
+			items = connectionNames.map(name => ({
+				label: name,
+				description: l10n.t(`Connection from account (${connections[name].account || 'No account specified'})`),
+			}));
+		} else {
+			// If no connections found, offer help options
+			items = [
+				{
+					label: l10n.t('Update path in settings'),
+					description: l10n.t('Create a connections.toml file in ~/.snowflake/ or set a custom path in settings.'),
+					detail: l10n.t('Configure the path to your Snowflake connections.toml file')
 				}
+			];
+		}
 
-				if (selection.label === 'Enter New Account...') {
-					// User wants to enter a new account
-					account = await vscode.window.showInputBox({
-						prompt: 'Enter your Snowflake account identifier',
-						placeHolder: 'orgname-accountname',
-						validateInput: (value) => {
-							return value.trim() === '' ? 'Account name cannot be empty' : null;
-						},
-						ignoreFocusOut: true
-					});
-				} else {
-					// User selected an existing account
-					account = selection.label;
-				}
-			} else {
-				// No recent accounts, just show the input box
-				account = await vscode.window.showInputBox({
-					prompt: 'Enter your Snowflake account identifier',
-					placeHolder: 'orgname-accountname',
-					validateInput: (value) => {
-						return value.trim() === '' ? 'Account name cannot be empty' : null;
-					},
-					ignoreFocusOut: true
-				});
-			}
+		const selection = await vscode.window.showQuickPick(items, {
+			placeHolder: 'Select a connection profile',
+			ignoreFocusOut: true
+		});
+
+		if (!selection) {
+			return undefined; // User canceled
 		}
 
-		// If still no account, user canceled
-		if (!account) {
+		if (selection.label === l10n.t('No connections.toml file found')) {
```

Comment:
Oops I think this needs to be updated

```suggestion
		if (selection.label === l10n.t('Update path in settings')) {
```